src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -11856,80 +11856,60 @@ void VRFCondenserEquipment::CalcVRFCondenser_FluidTCtrl(EnergyPlusData &state)
-                CyclingRatio = Q_c_OU * C_cap_operation / CompEvaporatingCAPSpdMin;
-                this->EvaporatingTemp = max(CapMinTe, RefTLow);
-            }
-
-            double CyclingRatioFrac = 0.85 + 0.15 * CyclingRatio;
-            double HPRTF = CyclingRatio / CyclingRatioFrac;
-            Ncomp = CompEvaporatingPWRSpdMin * HPRTF;
-            CompSpdActual = this->CompressorSpeed(1);
-
-        } else {
-            // Required heating load is greater than or equal to the min heating capacity
-
-            // Iteration_Ncomp: Perform iterations to calculate Ncomp (Label20)
-            Counter = 1;
-        Label20:;
-            Ncomp_new = Ncomp;
-            Q_c_OU = max(0.0, Q_h_TU_PL - Ncomp);
-
-            // *VRF OU Te calculations
-            m_air = this->OUAirFlowRate * RhoAir;
-            SH_OU = this->SH;
-            this->VRFOU_TeTc(
-                state, HXOpMode::EvapMode, Q_c_OU, SH_OU, m_air, OutdoorDryBulb, OutdoorHumRat, OutdoorPressure, Tfs, this->EvaporatingTemp);
-            this->SH = SH_OU;
-
-            // *VRF OU Compressor Simulation at heating mode: Specify the compressor speed and power consumption
-            this->VRFOU_CalcCompH(state,
-                                  TU_HeatingLoad,
-                                  this->EvaporatingTemp,
-                                  Tdischarge,
-                                  h_IU_cond_out_ave,
-                                  this->IUCondensingTemp,
-                                  CapMinTe,
-                                  Tfs,
-                                  Pipe_Q_h,
-                                  Q_c_OU,
-                                  CompSpdActual,
-                                  Ncomp_new);
-
-            if ((std::abs(Ncomp_new - Ncomp) > (Tolerance * Ncomp)) && (Counter < 30)) {
-                Ncomp = Ncomp_new;
-                Counter = Counter + 1;
-                goto Label20;
-            }
+        // Required heating load is greater than or equal to the min heating capacity
+
+        // Iteration_Ncomp: Perform iterations to calculate Ncomp (Label20)
+        Counter = 1;
+    Label20:;
+        Ncomp_new = Ncomp;
+        Q_c_OU = max(0.0, Q_h_TU_PL - Ncomp);
+
+        // *VRF OU Te calculations
+        m_air = this->OUAirFlowRate * RhoAir;
+        SH_OU = this->SH;
+        this->VRFOU_TeTc(state, HXOpMode::EvapMode, Q_c_OU, SH_OU, m_air, OutdoorDryBulb, OutdoorHumRat, OutdoorPressure, Tfs, this->EvaporatingTemp);
+        this->SH = SH_OU;
+
+        // *VRF OU Compressor Simulation at heating mode: Specify the compressor speed and power consumption
+        this->VRFOU_CalcCompH(state,
+                              TU_HeatingLoad,
+                              this->EvaporatingTemp,
+                              Tdischarge,
+                              h_IU_cond_out_ave,
+                              this->IUCondensingTemp,
+                              this->IUEvapTempLow,
```

Comment:
enforces a lower bound to the outdoor unit evaporating temperature

Reply:
I know the `goto Label20` was present in the original code but we should try to eliminate goto statements when possible. This does not need to change in this branch and is a suggestion for future code. I moved out of the loop any code that does not need to be calculated each iteration.

    Counter = 1;
    m_air = this->OUAirFlowRate * RhoAir;
    Ncomp_new = Ncomp;
    while (std::abs(Ncomp_new - Ncomp) > (Tolerance * Ncomp)) && (Counter < 30)) {
    }

Reply:
it feels more like a do-while loop, where the code inside label20 is executed once regardless of the condition, and then continues with a while loop. Although it is not a clean do-while loop as there's some operation inside the `if (condition) {goto label}` branch, especially for the label23 code, before goto, the `h_IU_cond_in` is recomputed differently. Let me think about it a bit.

Reply:
Would it be better to have a separate PR fixing goto ones? There are a few more instances in the physics-based VRF code. @rraustad

Reply:
The existing code is like the following after simplification. Let A = stuff A, B = stuff B and counter increment, C being the counter, then this code will be evaluated to something like the following regex: A(CBA)*C. 
```
label:
    stuff A
if (some condition) {
    stuff B
    goto label
}
```
A while loop like this will evaluate to C(BAC)*
```
while (condition) {
    stuff B
    stuff A
}
```
a do-while loop like this will evaluate to (BAC)+
```
do {
    stuff B
    stuff A
} while (condition)
```
Maybe there's not a uniform clean transition, we'll need to reorganize the original code a little bit (some might be straightforward others might not).

Reply:
That's fine to make goto changes in a different branch. I was just pointing that out.

Reply:
Definitely. I will have a separate branch for goto fixes so that they could be tested separately. I feel it might have been more involved as label20 is nested inside label23. They probably should be considered together while making changes.